src/auth/SignUp.tsx

```diff
@@ -0,0 +1,86 @@
+import { useForm, SubmitHandler } from 'react-hook-form';
+import axios from 'axios';
+import { useRef } from 'react';
+
+interface FormValue {
+  email: string;
+  fullName: string;
+  password: string;
+  passwordConfrim: string;
+}
+
+const SignUp = () => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isSubmitting, errors },
+  } = useForm<FormValue>();
+
+  const passwordRef = useRef<string | null>(null);
+  passwordRef.current = watch('password');
+
+  const onSubmitHandler: SubmitHandler<FormValue> = async (data) => {
+    await axios.post('http://kdt.frontend.3rd.programmers.co.kr:5006/signup', {
+      email: data.email,
+      fullName: data.fullName,
+      password: data.password,
+    });
+  };
```

Comment:
```suggestion
  const onSubmitHandler: SubmitHandler<FormValue> = async ({email, fullName, password}) => {
    await axios.post('http://kdt.frontend.3rd.programmers.co.kr:5006/signup', {
      email,
      fullName,
      password
    });
  };
```
비구조할당으로 꺼내서 쓰셔도 될 거 같아요~